webpack/scenes/ContentViews/Table/ContentViewsTable.js

```diff
@@ -3,29 +3,29 @@ import PropTypes from 'prop-types';
 import { translate as __ } from 'foremanReact/common/I18n';
 import { STATUS } from 'foremanReact/constants';
 
-import TableWrapper from './TableWrapper';
+import TableWrapper from '../../../components/Table/TableWrapper';
 import tableDataGenerator from './tableDataGenerator';
 import actionResolver from './actionResolver';
+import getContentViews from '../ContentViewsActions';
 
-const ContentViewTable = ({
-  items, status, error,
-}) => {
+const ContentViewTable = ({ response, status, error }) => {
   const [table, setTable] = useState({ rows: [], columns: [] });
   const [rowMapping, setRowMapping] = useState({});
   const loading = status === STATUS.PENDING;
+  const { results, ...metadata } = response;
 
   useEffect(
     () => {
-      if (!loading && items && items.length > 0) {
+      if (!loading && results && results.length > 0) {
```

Comment:
I think the issue with the flashing `EmptyStateMessage` is happening because `status` changes to `RESOLVED` before `rows` is populated.  This is while `tableDataGenerator` is running.

The second problem is that the `setRowMapping` in the `useEffect` in `ContentViewsTable.js` actually causes the same `useEffect` to run again, because `JSON.stringify(rowMapping)` is in the dependency array.  This means `tableDataGenerator` runs a second time, doubling the time on screen for the `EmptyStateMessage`.

I was playing around (for way too long) and I was able to prevent the `EmptyStateMessage` by comparing the size of `rowMapping` with `results.length`:

```js
const resultsLength = 0 || results && results.length
  const rowMappingLength = 0 || Object.keys(rowMapping) && Object.keys(rowMapping).length
  const displaySpinner = (status === STATUS.PENDING) || resultsLength !== rowMappingLength
  useEffect(
    () => {
      if (status = STATUS.PENDING && results && results.length > 0) {
...
```

I forced the status to `STATUS.PENDING` when rendering the table:
```js
return (
    <TableWrapper
      {...{
        rows,
        error,
        status: displaySpinner ? STATUS.PENDING : status,
...
```

This gets rid of the EmptyStateMessage flash.

There's probably a more elegant way to do it, but you get the idea.

To address the second problem, I think it would improve load time if you [memoize](https://reactjs.org/docs/hooks-reference.html#usememo) `tableDataGenerator`, since it's running multiple times with the same results data.

Reply:
Just realized my solution breaks when you change the per-page setting.  :shrug: